src/main/java/org/jabref/gui/BasePanel.java

```diff
@@ -767,12 +771,71 @@ private void cut() {
      *            "deleted". If true the action will be localized as "cut"
      */
     private void delete(boolean cut) {
-        List<BibEntry> entries = mainTable.getSelectedEntries();
+    	boolean bRemoveFromGroup = false;
+    	boolean bDelete = false;
+    	List<FieldChange> changesRemove = new ArrayList<>();
+    	
+    	List<BibEntry> entries = mainTable.getSelectedEntries();
+
         if (entries.isEmpty()) {
             return;
         }
-        if (!cut && !showDeleteConfirmationDialog(entries.size())) {
-            return;
+    	
+    	String selectedGroup = "";
+    	if (Globals.stateManager.getSelectedGroup(bibDatabaseContext).size() > 0) {
+    		selectedGroup = Globals.stateManager.getSelectedGroup(bibDatabaseContext).get(0).getName();
+    	}
+
+        //get list of all groups
+        Optional<GroupTreeNode> groups = bibDatabaseContext.getMetaData().getGroups();
+
+        if (groups.get().getName().contentEquals(groups.get().getName())) { //checks if is selected any group
+        	//TODO define dialog not to offer "remove from group"
+        }
+
+		Object[] options = {Localization.lang("Delete_from_database"),
+			Localization.lang("Remove_from_group") + " \"" + selectedGroup + "\"",
+			Localization.lang("Cancel")};
+		int n = JOptionPane.showOptionDialog(frame,
+				Localization.lang("The_entry_is_the_member_of_groups") + entries.get(0).getField("groups"),
```

Comment:
I think it would be best to put this dialog in a loop, because it it confusing if you delete many entries it displays only the group information from the first entry.
It get's displayed as Optional[Group1, Group2].  
You should add a check before the dialog: (assuming you have a loop)
```java
for(BibEntry entry:  entries)
{
  if (entry.getField("groups").isPresent())
  {
    show....
  }
}
```